### PR TITLE
Added chain select for wallet address form field

### PR DESCRIPTION
**Pull request by Devorein**
### WHY

[Card](https://app.charmverse.io/charmverse/extend-wallet-input-to-allow-a-blockchain-requirement-9395122206771407)

### CODE REVIEW

File: components/common/form/FormField.tsx [hunk 1]
```diff
@@ -218,6 +218,15 @@ function ExpandedFormField({
           onCreateOption={onCreateOption}
           onDeleteOption={onDeleteOption}
           onUpdateOption={onUpdateOption}
+          walletInputConfig={{
+            chainId: (formField.fieldConfig as { chainId: number } | undefined)?.chainId,
+            onChangeChainId: (chainId) => {
+              updateFormField({
+                fieldConfig: { chainId },
```
Comment (mattcasey) [inline components/common/form/FormField.tsx:225]: shouldn't this include existing field config if there is one?

File: components/common/form/FormFieldAnswers.tsx [hunk 2]
```diff
@@ -229,6 +229,13 @@ export function FormFieldAnswersControlled({
                     labelEndAdornment={
                       formField.private ? <Chip sx={{ ml: 1 }} label='Private' size='small' /> : undefined
                     }
+                    walletInputConfig={
+                      (formField.fieldConfig as { chainId: number })?.chainId
```
Comment (mattcasey) [inline components/common/form/FormFieldAnswers.tsx:233]: you should use a type guard, we dont use them a lot but you had to write the same code 3 times

File: components/common/form/fields/FieldTypeRenderer.tsx [hunk 3]
```diff
@@ -31,23 +38,44 @@ type SelectProps = {
 } & FieldProps &
   Required<ControlFieldProps>;
 
-type Props = (Omit<TextProps, 'type'> | Omit<SelectProps, 'type'>) & {
+type WalletProps = {
+  type: 'wallet';
+  walletInputConfig?: WalletInputConfig;
```
Comment (mattcasey) [inline components/common/form/fields/FieldTypeRenderer.tsx:43]: is walletInputConfig really undefined sometimes?

**Review by mattcasey — APPROVED** [review]

File: components/common/form/FormFieldAnswers.tsx (hunk 2 above)
Comment (Devorein) [reply components/common/form/FormFieldAnswers.tsx:233]: Good idea. Updated.

File: components/common/form/FormField.tsx (hunk 1 above)
Comment (Devorein) [reply components/common/form/FormField.tsx:225]: This is specifically for the wallet input config, so there won't be any existing field config

File: components/common/form/fields/FieldTypeRenderer.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (Devorein) [reply components/common/form/fields/FieldTypeRenderer.tsx:43]: We are making it optional in the Props underneath, this type isn't even necessary

**Pull request merged**